### DISCUSSION TITLE
Can I "press" a virtual Hubitat button from HA?

### DISCUSSION THREAD

**Discussion by euggersh** [jason0x43/hacs-hubitat#169] [Q&A] [answered]
I'm new to both Home Assistant and Hubitat so I hope this isn't an overly dumb question...

I've created a virtual button device in Hubitat which I use to trigger some rules in Rule Machine.  I've made this virtual button available in Maker API and it's found by the Hubitat integration in Home Assistant.

I see that I can use events from the virtual button to trigger automations in Home Assistant, but I'm wondering if I could have a Home Assistant automation virtually press the virtual button in Hubitat via the integration.

At the moment, I don't see any actions available when I select the virtual button as an action target in Home Assistant, but I don't know if that's because I have something misconfigured somewhere.

Is this possible to accomplish?  If not, is it a limitation of the integration or of the Maker API?

Thanks for any insight.

![image](https://user-images.githubusercontent.com/23174315/169927059-55d627a1-e641-48bc-82c9-43c0db6f7226.png)

**Comment by jason0x43** [comment, marked as answer]
It is possible, although it's currently a bit of a hack. Button controllers provided by the integration don't create Button entities (those didn't exist until relatively recently), so the built in `button.press` service won't work.

However, you can send arbitrary Hubitat commands using the `hubitat.send_command` service, as long as you have an entity to send commands ti. It can be any entity provide by the button controller device, such as a battery sensor. The tricky part is that pure button controller devices may not have any associated entities by default. If that's the case, you can enable the device's "last update time" entity (it's an entity that the integration can provide for every Hubitat device, but it's disabled by default).

Open your button controller in Settings -> Devices & Services -> Devices -> <find your device>. If you don't have anything under "sensors", click the "+1 entity not shown", then click the Last Update Time sensor, open the "Advanced Settings" dropdown, select "Enabled", then click "Update" at the bottom of the dialog.

<img width="643" alt="Screen Shot 2022-05-23 at 9 44 19 PM" src="https://user-images.githubusercontent.com/1042866/169931881-79f7801b-516b-4d82-a751-db51ad3fa26b.png">

<img width="619" alt="Screen Shot 2022-05-23 at 9 44 52 PM" src="https://user-images.githubusercontent.com/1042866/169932121-489a414e-2e59-4e90-a692-e3864184e690.png">

HA takes a few seconds to actually enable the device. Once it's enabled, you can go to Developer Tools -> Services, the try sending a "push" command to the button controller's last update time entity (or any other entity that exists for the device).

<img width="957" alt="Screen Shot 2022-05-23 at 9 48 23 PM" src="https://user-images.githubusercontent.com/1042866/169932271-e9a387ae-e6bf-4cd4-93ce-f1c4ecdb47ec.png">

**Comment by euggersh** [comment]
Thanks for the quick and detailed response!  I'll try this and see if I can get it figured out.

Edit:  It works!

**Reply by euggersh** [reply]
So, follow-up question... what are the chances of updating the integration to make it work with the new button-pressing functionality rather than the hack?

**Reply by jason0x43** [reply]
Decent, although I can't provide any sort of timeframe.
